lib/debug/thread_client.rb

```diff
@@ -348,10 +348,9 @@ def instance_eval_for_cmethod frame_self, src
     def frame_eval src, re_raise: false
       @success_last_eval = false
 
-      b = current_frame&.eval_binding || TOPLEVEL_BINDING
-      b = b.dup
```

Comment:
For some unknown reasons, I can't reproduce the polluted binding case anymore 🙈 . So I just remove the `.dup` call.

Reply:
Ah, it's because the current implementation only defines them at the frame that has those values. So it's not easy or even impossible to leak them into the program. 

For example, the `_return` is only defined at the end of the method definition, like 

```rb
def foo
 # ...
end # <= here
```

And theoretically no user program will be able to access it, so there's no need to worry about it polluting the program.

Reply:
without `.dup`, `TOPLEVEL_BINDING` will be affected with assignments.
Maybe it is not intended.